packages/fxa-auth-server/scripts/must-reset.js

```diff
@@ -20,68 +20,64 @@
 const commandLineOptions = require('commander');
-const config = require('../config').getProperties();
-const crypto = require('crypto');
-const log = require('../lib/log')({});
-const P = require('../lib/promise');
+const fs = require('fs');
+const main = require('./must-reset/index');
 const path = require('path');
-const Token = require('../lib/tokens')(log, config);
-const AuthDB = require('../lib/db')(config, log, Token);
-const oauth = require('../lib/oauth/db');
 
 commandLineOptions
-  .option('-i, --input <filename>', 'JSON input file')
+  .option('-e, --emails [emails]', 'Email addresses')
+  .option('-u, --uids [uids]', 'User IDs')
+  .option('-i, --input <filename>', 'Input filename from which to read input')
   .parse(process.argv);
 
-const requiredOptions = ['input'];
+if (!commandLineOptions.input) {
+  console.error(`-i, --input required`);
+  process.exit(1);
+} else if (!commandLineOptions.emails && !commandLineOptions.uids) {
+  console.error('One of `emails` or `uids` must be specified');
+  process.exit(1);
+} else if (commandLineOptions.emails && commandLineOptions.uids) {
+  console.error('Only one of `emails` or `uids` can be specified, not both');
+  process.exit(1);
+}
+
+let emails = [];
+let uids = [];
+
+if (commandLineOptions.emails) {
+  emails = getItems('emails');
+} else if (commandLineOptions.uids) {
+  uids = getItems('uids');
+}
 
-function checkRequiredOption(optionName) {
-  if (!commandLineOptions[optionName]) {
-    console.error(`--${optionName} required`);
+if (!emails.length && !uids.length) {
+  console.error('No `emails` or `uids` found inside the specified file');
+  process.exit(1);
+}
+
+function getItems(type) {
+  let input = '';
+  try {
+    input = fs
+      .readFileSync(path.resolve(commandLineOptions.input))
+      .toString('utf8');
+  } catch (err) {
+    console.error('No such file or directory');
     process.exit(1);
   }
+  return adjustText(input);
 }
 
-requiredOptions.forEach(checkRequiredOption);
-
-async function main() {
-  const db = await AuthDB.connect(config[config.db.backend]);
-
-  const json = require(path.resolve(commandLineOptions.input));
-
-  const uids = json.map((entry) => {
-    return entry.uid;
-  });
-
-  await P.all(
-    uids.map(async (uid) => {
-      try {
-        // Removes all session tokens,
-        await db.resetAccount(
-          { uid },
-          {
-            authSalt: butil.ONES.toString('hex'),
-            verifyHash: butil.ONES.toString('hex'),
-            wrapWrapKb: crypto.randomBytes(32).toString('hex'),
-            verifierVersion: 1,
-          }
-        );
-
-        // Removes oauth related tokens
-        await oauth.removeUser(uid);
-        console.info('account reset', uid);
-      } catch (err) {
-        console.error('failed', uid, err);
-        process.exit(1);
-      }
-    })
-  );
-
-  console.info('%s accounts reset', uids.length);
-  await db.close();
-  process.exit();
+function adjustText(input = '') {
```

Comment:
The function name is pretty ambiguous here, perhaps `splitCommaString`?

Reply:
`input` should also be more descriptive, perhaps `commaString` to demote its expected format?

Reply:
This function also splits white space, maybe another name?

Reply:
Ah, yes. How about a doc string on the function that indicates what kind of adjustment it does to the text?